### PR TITLE
Enhance VM Data (Longhorn) Partition feature

### PR DESCRIPTION
**Pull request by johnliu55tw**
- Add hard and soft disk size limit
- Add install config `NoDataPartition` (`install.no_data_partition`) to disable additional VM data partition. VM data will be sharing the partition with OS data if set to `true`.

## Hard and soft disk size requirements
- **Hard disk requirement: 60G.** If the selected disk is smaller than this size, installation cannot proceed.
- **Soft disk requirement: 140G.** If the selected disk is smaller than this size, Installer GUI will show a warning but users can still proceed to install onto the disk.

### Calculation of `COS_PERSISTENT` partition size

**NOTE: This is NOT the VM data partition!**

This is the partition for storing container images, etcd data, and everything that needs to be persisted after reboot.

- Size **greater than soft disk requirement**, i.e. `disk size >= 140G`: `COS_PERSISTENT` partition size will range from 50G to 100G, propositional to the disk size.
- Size **smaller than soft disk requirement but greater than hard disk requirement**, i.e. `60G <= disk size < 140G`: `COS_PERSISTENT` partition size will range from 25G to 49G, proportional to the disk size.

## Result of `NoDataPartition` and `ForceMBR` on UEFI/BIOS system
<table>
    <tbody>
        <tr>
            <td></td>
            <td></td>
            <td><b>(Default) NoDataPartition = false</b></td>
            <td><b>NoDataPartition = true</b></td>
        </tr>
        <tr>
            <td rowspan=2><b>UEFI</b></td>
            <td><b>ForceMBR = false</b></td>
            <td>GPT</td>
            <td>GPT without VM Data Partition</td>
        </tr>
        <tr>
            <td><b>ForceMBR = true</b></td>
            <td><i>ERR: UEFI can't force MBR</i></td>
            <td><i>ERR: UEFI can't force MBR</i></td>
        </tr>
        <tr>
            <td rowspan=2><b>BIOS</b></td>
            <td><b>ForceMBR = false</b></td>
            <td>GPT</td>
            <td>GPT without VM Data Partition</td>
        </tr>
        <tr>
            <td><b>ForceMBR = true</b></td>
            <td>MBR without VM Data Partition*</td>
            <td>MBR without VM Data Partition*</td>
        </tr>
    </tbody>
</table>

- **MBR without VM Data Partition***: Under BIOS system, if ForceMBR is `true`, we will never create a VM Data partition, ignoring the `NoDataPartition` setting.

## TODOs
- [x] Add option `NoVMDataPartition` to prohibit installer from creating a dedicated partition for Longhorn
- [x] Add soft-constraint for disk size
- [x] Add doc about the new option (Draft: https://github.com/harvester/docs/pull/82)

### CODE REVIEW

File: pkg/console/install_panels.go [hunk 1]
```diff
@@ -291,6 +276,14 @@ func addDiskPanel(c *Console) error {
 				return updateValidatorMessage(err.Error())
 			}
 
+			if err := validateDiskSizeSoft(device); err != nil && !userInputData.HasWarnedDiskSize {
+				userInputData.HasWarnedDiskSize = true
+				return updateValidatorMessage(fmt.Sprintf(
+					"Warning: %s.\nThis might cause system to become unstable. Press Enter continue.",
```
Comment (guangbochen) [inline pkg/console/install_panels.go:282]: probably showing the `Disk size smaller than recommended minimum size 140G. Presse Enter to continue.` is sufficient enough.

File: pkg/config/config.go [hunk 2]
```diff
@@ -71,7 +71,10 @@ type Install struct {
 	Debug     bool   `json:"debug,omitempty"`
 	TTY       string `json:"tty,omitempty"`
 	ForceGPT  bool   `json:"forceGpt,omitempty"`
-	ForceMBR  bool   `json:"forceMbr,omitempty"` // ForceMBR is not a cOS installer flag
+
+	// Following options are not cOS installer flag
+	ForceMBR          bool `json:"forceMbr,omitempty"`
+	NoVMDataPartition bool `json:"noVmDataPartition,omitempty"`
```
[the remark this answers is not in the document]
Comment (johnliu55tw) [reply pkg/config/config.go]: Renamed.

File: pkg/console/install_panels.go (hunk 1 above)
Comment (johnliu55tw) [reply pkg/console/install_panels.go]: Done

**Review by guangbochen — APPROVED** [review]
LGTM

**Comment by bk201** [comment]
The feature works well. But I feel the persistent partition might be too small for a 60 disk.

For a brand new-installed server:
```
# lsblk /dev/vda
NAME   MAJ:MIN RM  SIZE RO TYPE MOUNTPOINT
vda    253:0    0   60G  0 disk
├─vda1 253:1    0   47M  0 part
├─vda2 253:2    0   50M  0 part /oem
├─vda3 253:3    0   15G  0 part /run/initramfs/cos-state
├─vda4 253:4    0    8G  0 part
├─vda5 253:5    0   25G  0 part /usr/local
└─vda6 253:6    0 11.9G  0 part /var/lib/longhorn

# df /usr/local/ -h
Filesystem      Size  Used Avail Use% Mounted on
/dev/vda5        25G   19G  4.3G  82% /usr/local
```

There is only 4.3G left. Operations like upgrades need temporary space to store and extract tarballs, can we reserve more or drop the LH partition if the persistent space is too small?

**Comment by johnliu55tw** [comment]
> There is only 4.3G left. Operations like upgrades need temporary space to store and extract tarballs, can we reserve more or drop the LH partition if the persistent space is too small?

I'd say installing with the true minimum required size (60G) is for development only. It's expected to see the system behave abnormally with such a small disk. For example, Prometheus can't even start with such a small disk.

**Review by bk201 — APPROVED** [review]
LGTM

**Comment by guangbochen** [comment]
u probably need to run `git amend` to retrigger the CI, seems it was failed due to `url: https://192.168.0.30:30443`, this should already been fixed on the latest ansible script.

**Review by gitlawr — APPROVED** [review]
LGTM

**Pull request merged**